docs/machinepools.md

```diff
@@ -0,0 +1,85 @@
+# MachinePools
+- **Feature status:** Experimental
+- **Feature gate:** MachinePool=true
+
+MachinePool allows users to manage many machines as a single entity. Infrastructure providers 
+implement a separate CRD that handles infrastructure side of the feature. 
+
+
+## AWSMachinePool
+Cluster API Provider AWS (CAPA) has experimental support for `MachinePool` though the infrastructure
+type `AWSMachinePool`. An `AWSMachinePool` corresponds to an [AWS AutoScaling Groups](https://docs.aws.amazon.com/autoscaling/ec2/userguide/AutoScalingGroup.html), which provides the cloud provider specific resource for orchestrating a group of EC2 machines.
+
+The AWSMachinePool controller creates and manages an AWS AutoScaling Group using launch templates 
+so users don't have to manage individual machines. You can use Autoscaling health checks for replacing 
+instances and it will maintain the number of instances specified.
+
+
+### Using `clusterctl` to deploy
+To deploy a MachinePool / AWSMachinePool via `clusterctl config` there's a [flavor](https://cluster-api.sigs.k8s.io/clusterctl/commands/config-cluster.html#flavors)
+for that.
+
+Make sure to set up your AWS environment as described [here](https://cluster-api.sigs.k8s.io/user/quick-start.html).
+
+```shell
+export EXP_MACHINE_POOL=true
+clusterctl init --infrastructure aws
+clusterctl config cluster my-cluster --kubernetes-version v1.16.8 --flavor machinepool > my-cluster.yaml
+```
+
+The template used for this [flavor](https://cluster-api.sigs.k8s.io/clusterctl/commands/config-cluster.html#flavors)
+is located [here](../../templates/cluster-template-machinepool.yaml). For example you can add it to `~/.cluster-api/overrides/infrastructure-aws/0.0.0/cluster-template-machinepool.yaml`
```

Comment:
This wouldn't be needed if we add it to the release artifacts.....which is probably better. To do this i think the following is needed:

- update the [release-templates make target ](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/Makefile#L440) to copy the file
- update the [release docs](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/docs/book/src/development/releasing.md) to mention the template file

Reply:
I'll fix this as as a follow up.